src/app/settings/views/Configuration/Security/SessionTimeout/SessionTimeout.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+
+import { Spinner, Select, Button, Icon } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+
+type SessionTimeoutFormValues = {
+  session_length: number;
+  time_unit: "hours" | "days";
+};
+
+export enum Labels {
+  Loading = "Loading...",
+  Edit = "Edit",
+  Expiration = "Session timeout expiration",
+  Save = "Save",
+  Cancel = "Cancel",
+  TimeUnit = "Time unit",
+  Days = "Day(s)",
+  Hours = "Hour(s)",
+  ConfigureSessionTimeout = "Configure Session Timeout",
+}
+
+const SessionTimeout = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const configLoading = useSelector(configSelectors.loading);
+  const sessionLength = useSelector(configSelectors.sessionLength) || 1209600; // fallback to default value of 14 days if none is set
+  const saved = useSelector(configSelectors.saved);
+  const saving = useSelector(configSelectors.saving);
+
+  const [isFormOpen, setFormOpen] = useState(false);
+
+  const timeUnitOptions = [
+    { label: Labels.Hours, value: "hours" },
+    { label: Labels.Days, value: "days" },
+  ];
+
+  if (configLoading) {
+    return <Spinner aria-label={Labels.Loading} text={Labels.Loading} />;
+  }
+
+  if (!isFormOpen) {
+    return (
+      <>
+        <span className="u-text--muted">
+          Session timeout expiration
+          <Button
+            className="is-small p-button__show-form"
+            onClick={() => setFormOpen(true)}
+          >
+            <Icon name="edit" />
+            Edit
+          </Button>
+        </span>
+        <p>{sessionLength / 24 / 3600} days</p>
```

Comment:
It might be better to pass the value to one of date-fns utils that we already use extensively in maas-ui. It'll make it easy to include hours as well if have been set. https://date-fns.org/v2.29.3/docs/formatDuration

Reply:
Awesome, I added a function to timeSpan.ts to convert a duration to days using these functions